File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        int myarray [] = new int [k];

        Map <Integer, Integer> myMap = new HashMap<>();

        for (int item : nums){
            if (!myMap.containsKey(item)){
                myMap.put(item, 1);
            }

            else{
                myMap.put(item, myMap.get(item) + 1);
            }
        }

        for (int i = 0; i < k; i++){

            if (myMap.isEmpty()){
                break;
            }

            int max = (Collections.max(myMap.values()));
            int maxKey = 0;
            for (int key : myMap.keySet()){
                if (max == myMap.get(key)){
                    maxKey = key;
                }
            }

            myarray[i] = maxKey;
            myMap.remove(maxKey);
            
        }

        return myarray;
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        if len(nums) == 0:
            return len(nums)
        
        p1= 1
        p2 = 1

        while p2 < len(nums):

            if nums[p2] == nums[p2-1]:
                p2 = p2 + 1
            
            else:
                nums[p1] = nums[p2]
                p1 = p1 + 1
                p2 = p2 + 1
        
        return p1